### PR TITLE
Add edu.berkeley.BOINC

### PR DESCRIPTION
**Pull request by idunno101**
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I contacted upstream developers about submitting their software to Flathub. 
**Link:** https://github.com/BOINC/boinc/issues/2485#issuecomment-890457551
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
   The patch changes a couple of default switches when the app first launches.
   1. The boinc process will automatically shutdown when boincmgr is closed.
   2. The advanced GUI will be launched by default to avoid this 
   https://github.com/BOINC/boinc/issues/3415#issue-544302410

### CODE REVIEW

File: edu.berkeley.BOINC.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+app-id: edu.berkeley.BOINC
+runtime: org.gnome.Platform
+runtime-version: '40'
+sdk: org.gnome.Sdk
+command: boincmgr
+finish-args:
+  - --share=ipc
+  - --socket=x11
```
Comment (hfiguiere) [inline edu.berkeley.BOINC.yml:8]:
```suggestion
  - --socket=fallback-x11
```
Unless the app doesn't support wayland, in which case the wayland socket shall be removed

**Comment by hfiguiere** [comment]
bot, build edu.berkeley.BOINC

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: edu.berkeley.BOINC.yml (hunk 1 above)
Comment (idunno101) [reply edu.berkeley.BOINC.yml:8]: Right, the simulation graphics don't work on wayland so I guess I'll remove that.

**Comment by idunno101** [comment]
bot, build edu.berkeley.BOINC

**Comment by idunno101** [comment]
bot, build edu.berkeley.BOINC

**Comment by nedrichards** [comment]
bot, build edu.berkeley.BOINC

**Pull request closed without merging**